### PR TITLE
Unify two separate clock_getres and clock_gettime implementations. NFC

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: test/other/codesize/test_codesize_minimal_pthreads.jssize [hunk 1]
```diff
@@ -1 +1 @@
-8696
+8921
```
Comment (kripken) [inline test/other/codesize/test_codesize_minimal_pthreads.jssize:1]: Why did JS size increase here? The new musl code enabled here might be using a bunch of new JS perhaps?

Comment (sbc100) [reply test/other/codesize/test_codesize_minimal_pthreads.jssize]: I optimized the wasi clock JS code a little so now this regression is gone.

File: src/library_wasi.js [hunk 2]
```diff
@@ -136,41 +136,37 @@ var WasiLibrary = {
   },
 #endif
 
-  $checkWasiClock: (clock_id) => {
-    return clock_id == {{{ cDefs.__WASI_CLOCKID_REALTIME }}} ||
-           clock_id == {{{ cDefs.__WASI_CLOCKID_MONOTONIC }}} ||
-           clock_id == {{{ cDefs.__WASI_CLOCKID_PROCESS_CPUTIME_ID }}} ||
-           clock_id == {{{ cDefs.__WASI_CLOCKID_THREAD_CPUTIME_ID }}};
-  },
+  $checkWasiClock: (clock_id) => clock_id >= {{{ cDefs.__WASI_CLOCKID_REALTIME }}} && clock_id <= {{{ cDefs.__WASI_CLOCKID_THREAD_CPUTIME_ID }}},
```
Comment (kripken) [inline src/library_wasi.js:139]: Should we add a static assert here that these are in the order the code now assumes?

**Review by kripken — APPROVED** [review]

File: src/library_wasi.js (hunk 2 above)
Comment (sbc100) [reply src/library_wasi.js:139]: I would hope we can rely on test coverage for that.    (Also these really cannot change at this point).

File: test/other/codesize/test_codesize_minimal_pthreads.jssize (hunk 1 above)
Comment (kripken) [reply test/other/codesize/test_codesize_minimal_pthreads.jssize]: Great!

File: src/library_wasi.js (hunk 2 above)
Comment (kripken) [reply src/library_wasi.js:139]: Sounds good, lgtm.

**Comment by dschuff** [comment]
This seems to be failing under ASan: https://logs.chromium.org/logs/emscripten-releases/buildbucket/cr-buildbucket/8728960266806332305/+/u/Emscripten_testsuite__ASan_/stdout

**Pull request merged**